### PR TITLE
Add support for templates specifying dependencies on other templates

### PR DESCRIPTION
**Pull request by brikis98**
This PR updates `boilerplate` so that a `boilerplate.yml` file can specify one or more dependencies on other templates. This allows you to create more complicated boilerplate templates from simpler pieces.

### CODE REVIEW

File: examples/website/index.html [hunk 1]
```diff
@@ -4,6 +4,6 @@
   </head>
   <body>
     <h1>{{.WelcomeText}}</h1>
-    {{if .ShowLogo}}<img src="logo.png">{{end}}
+    {{if eq .ShowLogo "true"}}<img src="logo.png">{{end}}
```
Comment (brikis98) [inline examples/website/index.html:7]: This is a quick bug fix not related to this PR, but it annoyed me enough that I had to correct it :)

**Comment by brikis98** [comment]
@josh-padnick This is ready for review. And... it's awesome. I'll show you how I'm using this in another PR soon :)

**Comment by brikis98** [comment]
@josh-padnick Scratch that, don't merge yet. There is one key feature missing in this PR: the ability to namespace variables for the cases where multiple dependencies use the same variable name, but need different values. I'll come back to this a bit later today and fix it.

**Comment by brikis98** [comment]
@josh-padnick OK, we now have support for handling duplicate variable names between dependencies by using the <DEPENDENCY_NAME>.<VARIABLE_NAME> syntax. Review away!

**Comment by brikis98** [comment]
Merging this one in.

**Comment by josh-padnick** [comment]
Just getting to this PR from 23 days ago. :( I'll pass on reviewing unless there's anything in particular you'd like feedback on.

**Pull request merged**